### PR TITLE
Rc<RefCell<Gameboy>>

### PR DESCRIPTION
**Pull request by diogotito**
I know this isn't the agreed upon way to share the Gameboy between the GUI and the Debugger.
But look at this UI!

https://github.com/pineman/fpt/assets/3079989/417c1df7-6baa-491c-8273-88dd5afb43a0

(sorry for the light background)

### CODE REVIEW

File: fpt-egui/src/main.rs [hunk 1]
```diff
@@ -213,100 +228,205 @@ impl FPT {
...
+                    });
+                });
+                ui.add_space(20.0);
+                ui.horizontal(|ui| {
+                    ui.heading("Debugger:");
+                    if ui.button("Pause").clicked() {
+                        self.paused = true;
+                    }
+                    if ui.button("Step").clicked() {
+                        self.gb().frame();
```
Comment (pineman) [inline fpt-egui/src/main.rs]: this doesn't make much sense. sure, it updates some of the gameboy's state but it doesn't draw the frame or anything else. it also wont work when with interrupts, input we'll have to feed into the gameboy, audio, ...

I suggest we remove it for now and think of a better way of implementing it across egui updates

**Review by pineman — APPROVED** [review]

**Comment by pineman** [comment]
nice!!!!!!
looks like we'll have to keep arc mutex 😆

File: fpt-egui/src/main.rs (hunk 1 above)
Comment (diogotito) [reply fpt-egui/src/main.rs]: yeah, I should've called it "Next frame". But I didn't give much thought about the function of the buttons yet, I only wrote them to see where I liked them best in the UI, as an excuse to play with egui layouts (speaking of which, I totally forgot to showcase the Registers collapsible in the video)

**Comment by diogotito** [comment]
Just to clear my conscience, these changes are a bit hacky and other approaches (`trait Gameboy`, putting the `Debugger` inside the `Gameboy`) are still on the table

**Comment by pineman** [comment]
I say we move forward with this for now, since we still have so much other stuff to do, and revisit it later if need be

**Comment by diogotito** [comment]
Now it's `Rc<RefCell<Gameboy>>` because Clippy told so

**Pull request merged**